File: 26.remove-duplicates-from-sorted-array/solution.cpp
class Solution {
public:
	int removeDuplicates(vector<int>& nums) {
		if (nums.empty()) return 0;
		int result = 1;
		vector<int>::iterator cur = nums.begin();
		for (vector<int>::const_iterator it = nums.begin() + 1; it != nums.end(); it++)
		{
			if (*cur != *it)
			{
				result++;
				*(++cur) = *it;
			}
		}
		return result;
	}
};

File: 26.remove-duplicates-from-sorted-array/solution.kt
class Solution {
    fun removeDuplicates(nums: IntArray): Int {
        var j = 0
        for (i in (1 until nums.size)) {
            if (nums[i] != nums[j]) nums[++j] = nums[i]
        }
        return j + 1
    }
}
